app/src/main/java/com/kamron/pogoiv/logic/PokemonNameCorrector.java

```diff
@@ -41,54 +41,53 @@ public void putCorrection(String ocredPokemonName, String correctedPokemonName)
      * Gets the best matching pokemon that can be found given the input, by doing the following:
      * 1. check if the nickname perfectly matches a pokemon
      * 2. check if candyname + evolution cost perfectly matches a pokemon
-     * 3. check if there's a cached result for the scanned pokemon name
-     * 4. get the pokemon with the closest name within the evolution line guessed from the candy
+     * 3. check if there's a cached result for the nickname bitmap hash
+     * 4. check if there's a stored user correction for the scanned pokemon name
+     * 5. get the pokemon with the closest name within the evolution line guessed from the candy
+     * <p>
+     * The order is decided by having high reliability guessing modules run first, and if they cant find an answer,
+     * fall back to less accurate methods.
      *
      * @param poketext         the scanned pokemon nickname
      * @param candytext        the scanned pokemon candy name
      * @param candyUpgradeCost the scanned pokemon evolution candy cost
      * @return a Pokedist with the best guess of the pokemon
      */
     public PokeDist getPossiblePokemon(String poketext, String candytext, Optional<Integer> candyUpgradeCost) {
-        /* If the user previous corrected this text, go with that. */
-        if (userCorrections.containsKey(poketext)) {
-            poketext = userCorrections.get(poketext);
-        }
-
-        //1. if there's a cached result for the nickname, return that
-        /* Go with the cached value; if no entry is found, returning null is correct. */
-        PokeDist cacheGuess = cachedCorrections.get(poketext);
-        if (cacheGuess != null) {
-            return cacheGuess;
-        }
-
         ArrayList<Pokemon> bestGuessEvolutionLine = getBestGuessForEvolutionLine(candytext);
-        PokeDist guess;
+        PokeDist guess = null;
         //Warning: Don't forget to call cacheResult before exiting. Preferably keep a single return point.
```

Comment:
The new code violates this comment—the code I think is fine, can you update the comment?